packages/calypso-storybook/src/index.js

```diff
@@ -0,0 +1,65 @@
+const calypsoBabelConfig = require( '@automattic/calypso-babel-config' );
+
+module.exports = function storybookDefaultConfig( {
+	stories,
+	plugins = [],
+	webpackAliases = {},
+	babelCacheDirectory,
+} = {} ) {
+	return {
+		core: {
+			builder: 'webpack5',
+		},
+		features: {
+			/**
+			 * Can probably be removed after the next major storybook release with emotion 11 support.
+			 *
+			 * @see https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#emotion11-quasi-compatibility
+			 */
+			emotionAlias: false,
+			babelModeV7: true,
+		},
+
+		babel: async ( storybookConfig ) => {
+			const baseConfig = calypsoBabelConfig();
+			return {
+				...storybookConfig,
+				// Cache is set via parameters, as each consumer of this plugin may want a different folder
+				// and we certainly don't want to harcode paths here.
+				...( babelCacheDirectory ? { cacheDirectory: babelCacheDirectory } : {} ),
+
+				// Reuse presets and plugins from the base Calypso config
+				presets: [ ...( storybookConfig.presets ?? [] ), ...( baseConfig.presets ?? [] ) ],
+				plugins: [
+					...( storybookConfig.plugins ?? [] ),
+					...( baseConfig.plugins ?? [] ),
+
+					// Forces some plugins to load in loose mode, used by Storybook.
+					// See https://github.com/storybookjs/storybook/issues/14805
+					[
+						require.resolve( '@babel/plugin-proposal-private-property-in-object' ),
+						{ loose: true },
+					],
+					[ require.resolve( '@babel/plugin-proposal-class-properties' ), { loose: true } ],
+					[ require.resolve( '@babel/plugin-proposal-private-methods' ), { loose: true } ],
+				],
+			};
+		},
+
+		stories: stories && stories.length ? stories : [ '../src/**/*.stories.{js,jsx,ts,tsx}' ],
```

Comment:
Because this file is imported in `<package>/.storybook/index.js`, so we need `<package>/.storybook/../src`. Maybe we can pass `rootDir` to this `storybookDefaultConfig` function and compose it? Current form is not very intuitive.